compiler/modules/front/resolveTypes.bal

```diff
@@ -329,6 +338,18 @@ function resolveTypeDesc(ModuleSymbols mod, s:ModuleLevelDefn modDefn, int depth
     panic error("unimplemented type-descriptor");
 }
 
+function nonEmptyType(ModuleSymbols mod, s:ModuleLevelDefn modDefn, s:TypeDesc td, t:SemType ty) returns t:SemType|ResolveTypeError {
+    final t:Env env = mod.tc.env;
+    d:Location loc = s:locationInDefn(modDefn, { startPos: td.startPos, endPos: td.endPos });
+    if !env.isReady() {
+        mod.possiblyEmptyTypes.push([ty, loc]);
+    }
+    else if t:isEmpty(mod.tc, ty) {
```

Comment:
We should factor these three lines out, because they are used in two places. We don't want the same error message in two places.

Reply:
Refactored to `testTypeForFiniteShape`. (Used that name since even though we are only checking for emptiness error is more specific)